src/validation/schema.js

```diff
@@ -322,13 +378,203 @@ const validateEntityFields = (defs, def) =>
     List(),
   )
 
+const validateNoImportsDirective = def =>
+  def.directives.find(directive => directive.name.value == 'imports')
+    ? List().push(
+        immutable.fromJS({
+          loc: def.name.loc,
+          entity: def.name.value,
+          message: `@imports directive only allowed on '${RESERVED_TYPE}' type`,
+        }),
+      )
+    : List()
+
+const importDirectiveTypeValidators = {
+  StringValue: (_def, _directive, _type) => List(),
+  ObjectValue: (def, directive, type) => {
+    let errors = List()
+    if (type.fields.length != 2) {
+      return errors.push(
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message:
+            'Imported type must be one of "Name", { name: "Name" }, or { name: "Name", as: "Alias" }',
+        }),
+      )
+    }
+    return type.fields.reduce((errors, field) => {
+      if (!['name', 'as'].includes(field.name.value)) {
+        return errors.push(
+          immutable.fromJS({
+            loc: directive.name.loc,
+            entity: def.name.value,
+            message: `@imports type object field '${field.name.value}' invalid,  may only be one of: [name, as]`,
```

Comment:
`@imports` -> `@import`, remove the `type object`, remove the double space after `invalid,`